internal-services/catalog/file-updates-process-file-updates-task.yaml

```diff
@@ -118,11 +125,44 @@ spec:
                   valueSize="$(yq "${key}" ${targetFile} |wc -l)"
                   startBlock=$(( ${foundAt} + ${blankLinesBeforeYaml} ))
 
+                  # the replacement should be a sed expression using "|" as separator
+                  if [[ $(tr -dc "|" <<< "${replacement}" | wc -m ) != 3 ]]; then
+                      updateError='Replace expression should be in `|search|replace|` format'
+                      break
+                  fi
+
                   # run the replace
                   sed -i "${startBlock},+${valueSize}s${replacement}" "${targetFile}"
+
+                  # get the replace part of "|search|replace|"
+                  replaceStr=$(awk -F"|" '{print $3}' <<< "${replacement}")
+
+                  # when the value is a text block we must make sure
+                  # only a single line was replaced and that the result
+                  # block has the same number of lines as before
+                  replacedBlockLines=$(sed -ne "${startBlock},+${valueSize}p" "${targetFile}" |wc -l)
+                  if [[ $replacedBlockLines != $(( $valueSize +1 )) ]]; then
+                      updateError="Text block size differs from the original"
+                      break
+                  fi
+
+                  # check if only a single line was replaced
+                  replacedCount=$(sed -ne "${startBlock},+${valueSize}p" "${targetFile}" |grep "${replaceStr}"  |wc -l)
+                  if [[ $replacedCount != 1 ]]; then
+                      updateError="Too many lines replaced. Check the replace expression"
```

Comment:
do you think we can show the user what we attempted to replace? Do we know at this point?

Reply:
We have both the replace expression and the result. The only issue is that this value would need to be saved in the result. Depending on the size of the output, it would fail the PipelineRun.